src/orange/detail/UnitInserter.cc

```diff
@@ -338,6 +351,18 @@ VolumeRecord UnitInserter::insert_volume(const SurfacesRecord& surf_record,
     return output;
 }
 
+void UnitInserter::process_daughter(VolumeRecord&              vol_record,
+                                    std::vector<Translation>   translations,
```

Comment:
This parameter is mistakenly by value (oops!)

Also in Celeritas to make it obvious from a calling site that an argument is expected to be modified, we pass raw pointers rather than mutable references: this was the recommendation of the google style guide up to early 2020 (I just now found out it's changed). The practice would have made it obvious here that a reference is missing on the `translations` argument.